panic when no resources are set

When no resources are set when creating a `Prometheus` TPR then the operator panics as it is trying to set a resource request on a nil map.

Logs:

```
panic: assignment to entry in nil map

goroutine 21 [running]:
panic(0xea0f00, 0xc42046e480)
        /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
github.com/coreos/prometheus-operator/pkg/operator.makePetSet(0xc42042c4f0, 0xa, 0xc4203ae8c0, 0x1e, 0xc42042c500, 0xf, 0x0, 0x0, 0xc42042c510, 0xa, ...)
        /Users/fredericbranczyk/go/src/github.com/coreos/prometheus-operator/pkg/operator/petset.go:46 +0x8ec
github.com/coreos/prometheus-operator/pkg/operator.(*Operator).reconcile(0xc4200855f0, 0xc4200576d0, 0xc420027f90, 0x1058b01)
        /Users/fredericbranczyk/go/src/github.com/coreos/prometheus-operator/pkg/operator/operator.go:421 +0xfcc
github.com/coreos/prometheus-operator/pkg/operator.(*Operator).worker(0xc4200855f0)
        /Users/fredericbranczyk/go/src/github.com/coreos/prometheus-operator/pkg/operator/operator.go:286 +0x82
created by github.com/coreos/prometheus-operator/pkg/operator.(*Operator).Run
        /Users/fredericbranczyk/go/src/github.com/coreos/prometheus-operator/pkg/operator/operator.go:103 +0xa0
```

Should be a simple fix by just instantiating a map if it is `nil`.

@fabxc

--- a/pkg/operator/petset.go
+++ b/pkg/operator/petset.go
@@ -42,6 +42,10 @@ func makePetSet(p spec.Prometheus, old *v1alpha1.PetSet, alertmanagers []string)
 	if p.Spec.Retention == "" {
 		p.Spec.Retention = "24h"
 	}
+
+	if p.Spec.Resources.Requests == nil {
+		p.Spec.Resources.Requests = v1.ResourceList{}
+	}
 	if _, ok := p.Spec.Resources.Requests[v1.ResourceMemory]; !ok {
 		p.Spec.Resources.Requests[v1.ResourceMemory] = resource.MustParse("2Gi")
 	}

